src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -59,370 +60,387 @@ module.exports = function<T, P, I, TI, C>(
     updateClassInstance,
   } = ReactFiberClassComponent(scheduleUpdate);
 
-  function markChildAsProgressed(current, workInProgress, priorityLevel) {
-    // We now have clones. Let's store them as the currently progressed work.
-    workInProgress.progressedChild = workInProgress.child;
-    workInProgress.progressedPriority = priorityLevel;
-    if (current) {
-      // We also store it on the current. When the alternate swaps in we can
-      // continue from this point.
-      current.progressedChild = workInProgress.progressedChild;
-      current.progressedPriority = workInProgress.progressedPriority;
+  function BeginWork(forceReplace : boolean) {
```

Comment:
The changes in this file are minimal; there's a lot of noise because I wrapped everything in this function, which increased the indentation.

The only real change is the addition of a `forceReplace` parameter.